content/tutorial/tutorial.md

```diff
@@ -1042,56 +1049,56 @@ Let's `map` over the `history` in the Game's `render` method:
 
-`key` is a special and reserved property in React (along with `ref`, a more advanced feature). When an element is created, React extracts the `key` property and stores the key directly on the returned element. Even though `key` may look like it belongs in `props`, `key` cannot be referenced using `this.props.key`. React automatically uses `key` to decide which components to update. A component cannot inquire about its `key`.
+`key` React की एक विशेष और रिजर्व्ड property है (`ref` के साथ, जो की और भी एडवांस्ड फीचर है)। जब React में एक एलिमेंट बनता है, React `key` property को निकालता है और रिटर्न्ड एलिमेंट में सीधे स्टोर करता है। भले ही लगे की `key` props के साथ है, लेकिन `key` को `this.props.key` से रिफरेन्स नहीं कर सकते। React अपने आप `key` को इस्तेमाल करता है और तय करता है की कोनसा कौम्पोनॅन्ट अपडेट करना है। कौम्पोनॅन्ट अपनी `key` के बारे में जांच नहीं कर सकता।     
 
-**It's strongly recommended that you assign proper keys whenever you build dynamic lists.** If you don't have an appropriate key, you may want to consider restructuring your data so that you do.
+**यह दृढ़ता से अनुशंसा की जाती है कि जब भी आप डायनामिक सूचियों का निर्माण करते हैं तो आप उचित key जरूर प्रदान करे** यदि आपके पास एक उपयुक्त key नहीं है, तो आप डाटा को restructure करने का सोचें, ताकि key आपके पास रहे।
 
-If no key is specified, React will present a warning and use the array index as a key by default. Using the array index as a key is problematic when trying to re-order a list's items or inserting/removing list items. Explicitly passing `key={i}` silences the warning but has the same problems as array indices and is not recommended in most cases.
+यदि कोई कुंजी निर्दिष्ट नहीं है, तो React एक चेतावनी पेश करेगा और डिफ़ॉल्ट रूप से array इंडेक्स को कुंजी के रूप में उपयोग करेगा। ऐरे इंडेक्स को `key` की तरह इस्तेमाल करना लिस्ट आइटम re-order या डालना / हटाना समस्यात्मक हो सकता है। स्पष्ट रूप से `key={i}` चेतावनी को शांत करता है लेकिन ऐरे इंडिसिस वाली ही समस्या रहती है और ज़्यादातर केसेस में ये रेकमेंडेड नहीं है।
 
-Keys do not need to be globally unique; they only need to be unique between components and their siblings.
+key को विश्व स्तर पर अद्वितीय होने की आवश्यकता नहीं है; उन्हें केवल कौम्पोनॅन्ट और उनके सिब्लिंग्स के बीच अद्वितीय होना चाहिए।
 
 
-### Implementing Time Travel {#implementing-time-travel}
+### टाइम ट्रेवल लागू करना {#implementing-time-travel}
 
-In the tic-tac-toe game's history, each past move has a unique ID associated with it: it's the sequential number of the move. The moves are never re-ordered, deleted, or inserted in the middle, so it's safe to use the move index as a key.
+टिक-टैक-टो खेल के इतिहास में, प्रत्येक पिछले चाल में एक अद्वितीय आईडी जुड़ी होती है: यह चाल की क्रमिक संख्या होती है। चालों को फिर से कभी भी आदेश नहीं दिया जाता है, हटा दिया जाता है, या बीच में डाला जाता है, इसलिए यह key के रूप में चाल सूचकांक का उपयोग करना सुरक्षित है।
```

Comment:
चालों को फिर से कभी भी आदेश नहीं दिया जाता है, हटा दिया जाता है, या बीच में डाला जाता है, इसलिए यह key के रूप में चाल सूचकांक का उपयोग करना सुरक्षित है। -> revisit?

Reply:
```
सारी चालें कभी भी re-order, हटाइ या बीच में डाली नहीं जाती है, तो चाल के इंडेक्स को key की तरफ इस्तेमाल करना सुरक्षित है
```

changed to this